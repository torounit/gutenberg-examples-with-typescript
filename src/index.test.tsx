import React from 'react'
import { create } from 'react-test-renderer'
import { BlockEditProps, BlockSaveProps } from '@wordpress/blocks';
import {
    Save, Edit, BlockAttributes
} from './index'

describe("index.tsx", () => {
    beforeAll(() => {
        window.matchMedia = jest.fn()
    })
    describe('save', () => {
        const props: BlockSaveProps<BlockAttributes> = {
            attributes: {
                content: 'content'
            },
        }
        it("should match snapshot", () => {
            expect(create(<Save {...props} />).toJSON()).toMatchSnapshot()
        })
    })
    describe("Edit", () => {
        const props: BlockEditProps<BlockAttributes> = {
            attributes: {
                content: 'content'
            },
            isSelected: false,
            className: '',
            setAttributes: jest.fn(),
        }
        it("should match snapshot", () => {
            expect(create(<Edit {...props} />).toJSON()).toMatchSnapshot()
        })
    })
})