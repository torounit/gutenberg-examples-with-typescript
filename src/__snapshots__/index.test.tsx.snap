// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.tsx Edit should match snapshot 1`] = `
<p>
  hello
</p>
`;

exports[`index.tsx save should match snapshot 1`] = `
<p>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "content",
      }
    }
  />
</p>
`;
